protocol/binary/stream_reader_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package binary
+
+import (
+	"bytes"
+	"io"
+	"testing"
+)
+
+func BenchmarkReadString(b *testing.B) {
+	var streamBuff bytes.Buffer
+
+	// Encode with Streaming protocol
+	w := NewStreamWriter(&streamBuff)
+	w.WriteString("the quick brown fox jumps over the lazy dog")
+	w.Close()
+
+	encoded := streamBuff.Bytes()
+
+	sr := new(StreamReader)
+	enc := bytes.NewReader(encoded)
+
+	b.ResetTimer()
+
+	for i:= 0; i<b.N; i++ {
+		resetStreamReader(sr, enc)
```

Comment:
what's the role of resetStreamReader? Would it also work to set the reader field once since the same encoder is being used?

Would managing the io.Seeker best be done here as well?

Reply:
Yes, that would be cleaner. Thanks for the suggestion.